modules/setting/history/exposed_api.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package history
+
+// This file contains the methods that will be exposed to other packages
+import (
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting/base"
+)
+
+var removedSettings map[SettingsSource]map[string][]*historyEntry = make(map[SettingsSource]map[string][]*historyEntry) // ordered by old source and then by old section (for performance)
```

Comment:
Should we really need to store this in the memory, once we release Gitea v100, it will become very big. :) I think we just need to print them in console or log?

Reply:
Or we can destroy it after printing.

Reply:
1. I would have guessed that the Go GC takes care of it once `PrintRemovedSettings` has been called, as that package isn't referenced from anything anymore then.
2. Even if the Go GC fails to do its duty, I doubt we will ever reach noticeable sizes: Let's assume one entry uses about 200 Bytes (which is already an overestimation), then we still need 5 settings for 1KB, or 5 000 settings for 1MB.

But yes, we can of course clear the map afterward manually.

Reply:
If a package is referenced once, it's always referenced, Go never de-references / unloads packages.

GC doesn't GC global variables.

Reply:
> 1. I would have guessed that the Go GC takes care of it once `PrintRemovedSettings` has been called, as that package isn't referenced from anything anymore then.
> 
>     2. Even if the Go GC fails to do its duty, I doubt we will ever reach noticeable sizes: Let's assume one entry uses about 200 Bytes (which is already an overestimation), then we still need 5 settings for 1KB, or 5 000 settings for 1MB.
> 
> 
> But yes, we can of course clear the map afterward manually.

Of course, I know it's small but I want to save every byte when possible.